presto-main/src/main/java/com/facebook/presto/split/BufferingSplitSource.java

```diff
@@ -45,6 +45,12 @@ public ConnectorId getConnectorId()
     }
 
     @Override
+    public Split getEmptySplit()
```

Comment:
Yeah, it does look weird. Empty splits are purely a construct needed by the engine (planner/scheduler), so it seems very unnatural for SplitSource to have to be aware of it. 

What's so hard about getting them to SourcePartitionedScheduler? It'd be cleaner for it to just take an "emptySplitSource" in addition to the regular split source. It could probably be done by wrapping `Map<PlanNodeId, SplitSource> splitSources` in an object that contains both regular splitSources and an emptySplit source. Wherever the regular sources are created, it should be possible to create the empty split source.